ooniprobe/Model/Database/Result.mm

```diff
@@ -90,8 +83,25 @@ -(NSString*)getLocalizedNetworkType{
     return @"";
 }
 
--(void)addRuntime:(float)value{
-    self.runtime+=value;
+-(Measurement*)getFirstMeasurement{
+    SRKResultSet *measurements = [[[[Measurement query] where:@"result_id = ? AND is_rerun = 0" parameters:@[self]] order:@"start_time"] fetch];
+    if ([measurements count] > 0)
+        return [measurements objectAtIndex:0];
+    return nil;
+}
+
+-(Measurement*)getLastMeasurement{
+    SRKResultSet *measurements = [[[[Measurement query] where:@"result_id = ? AND is_rerun = 0" parameters:@[self]] orderByDescending:@"start_time"] fetch];
+    if ([measurements count] > 0)
+        return [measurements objectAtIndex:0];
+    return nil;
+}
+
+-(float)getRuntime{
+    Measurement *first = [self getFirstMeasurement];
+    Measurement *last = [self getLastMeasurement];
+    NSTimeInterval secondsBetweenTests = [last.start_time timeIntervalSinceDate:first.start_time];
+    return secondsBetweenTests + first.runtime;
```

Comment:
You should be summing the runtime of the **last** run to get the total runtime.

Reply:
Correct